src/src_python/antares_xpansion/driver.py

```diff
@@ -262,29 +262,38 @@ def pre_antares(self):
-        if self.is_accurate():
-            ini_file['general']['mode'] = 'expansion'
-            ini_file['other preferences']['unit-commitment-mode'] = 'accurate'
-            ini_file[self.config.OPTIMIZATION]['include-tc-minstablepower'] = 'true'
-            ini_file[self.config.OPTIMIZATION]['include-tc-min-ud-time'] = 'true'
-            ini_file[self.config.OPTIMIZATION]['include-dayahead'] = 'true'
-        else:
-            ini_file['general']['mode'] = 'Economy'
-            ini_file['other preferences']['unit-commitment-mode'] = 'fast'
-            ini_file[self.config.OPTIMIZATION]['include-tc-minstablepower'] = 'false'
-            ini_file[self.config.OPTIMIZATION]['include-tc-min-ud-time'] = 'false'
-            ini_file[self.config.OPTIMIZATION]['include-dayahead'] = 'false'
-
-        with open(self.general_data(), 'w') as out_file:
-            ini_file.write(out_file)
+        
+        changed_val = {'[' + self.config.OPTIMIZATION + ']/' + self.config.EXPORT_MPS       : 'true',
+                       '[' + self.config.OPTIMIZATION + ']/' + self.config.EXPORT_STRUCTURE : 'true',
+                       '[' + self.config.OPTIMIZATION + ']/' + 'include-tc-minstablepower'  : 'true' if self.is_accurate() else 'false',
+                       '[' + self.config.OPTIMIZATION + ']/' + 'include-tc-min-ud-time'     : 'true' if self.is_accurate() else 'false',
+                       '[' + self.config.OPTIMIZATION + ']/' + 'include-dayahead'           : 'true' if self.is_accurate() else 'false',
+                       '[' + self.config.OPTIMIZATION + ']/' + self.config.USE_XPRS         : '',
+                       '[' + self.config.OPTIMIZATION + ']/' + self.config.INBASIS          : '',
+                       '[' + self.config.OPTIMIZATION + ']/' + self.config.OUTBASIS         : '',
+                       '[' + 'general'                + ']/' + 'mode'                       : 'expansion' if self.is_accurate() else 'Economy',
+                       '[' + 'other preferences'      + ']/' + 'unit-commitment-mode'       : 'accurate' if self.is_accurate() else 'fast'
+                       }
+                       
+        with open(self.general_data(), 'r') as reader:
+            lines = reader.readlines()
+        
+        section = ""
+        with open(self.general_data(), 'w') as writer:
+            for line in lines:
+                lineWrite = line
+                split_str = line.split('=')
+                if (len(split_str) == 2 ):
+                    key = split_str[0].strip()
+                    value = split_str[1].strip()
+                    dict_key = section +'/'+key
+                    if (dict_key in changed_val):
+                        new_val = changed_val[dict_key]
+                        if (new_val):
+                            lineWrite = key + '=' + new_val + '\n'
+                else:
+                    section = line.strip()
+                writer.write(lineWrite)
```

Comment:
You want to strip only '\n'. Section names might contain white spaces, we don't want to remove them.